package study.guidanceOfJava11.Error.TryCatch;

//an example of nested try statements
class NestTry {
    public static void main(String[] args) {
        try{
            int a = args.length;
            /*
            if no command-line args aer present,
            the following statement will generate
            a divide-by-zero expression. */
            int b = 42 / a;

            System.out.println("a = " + a);

            try{
                //nested try block
                /*
                if one command-line arg is used,
                then a divide-by-zero expression
                will be generated by the following code
                 */
                if(a == 1){
                    a = a/(a-a);
                }
                /*
                if the two command-lines are used,
                then generate an out-of-bound expression
                 */
                if(a == 2){
                    int c[] = {1};
                    //generate an out-of-bound expression
                    c[42] = 99;
                }
            }catch (ArrayIndexOutOfBoundsException e){
                System.out.println("Array index out of bound: " + e);
            }
        }catch (ArithmeticException e){
            System.out.println("Divided by zero:" + e);
        }
    }
}
